dspace-api/src/main/java/org/dspace/core/HibernateDBConnection.java

```diff
@@ -203,44 +206,38 @@ private void configureDatabaseMode() throws SQLException {
                         uncacheEntity(bitstream);
                     }
                 }
-            //} else if(entity instanceof Bitstream) {
-            //    Bitstream bitstream = (Bitstream) entity;
-            //    No specific child entities to decache
+
+            // BITSTREAM
+            // No specific child entities to decache
+
+            // COMMUNITY
             } else if (entity instanceof Community) {
                 Community community = (Community) entity;
-                if(Hibernate.isInitialized(community.getAdministrators())) {
-                    uncacheEntity(community.getAdministrators());
-                }
+
+                //We don't uncache groups as they might still be referenced from the Context object
 
                 if(Hibernate.isInitialized(community.getLogo())) {
                     uncacheEntity(community.getLogo());
                 }
+
+            // COLLECTION
             } else if (entity instanceof Collection) {
                 Collection collection = (Collection) entity;
+
+                //We don't uncache groups as they might still be referenced from the Context object
+
                 if(Hibernate.isInitialized(collection.getLogo())) {
                     uncacheEntity(collection.getLogo());
                 }
-                if(Hibernate.isInitialized(collection.getAdministrators())) {
-                    uncacheEntity(collection.getAdministrators());
-                }
-                if(Hibernate.isInitialized(collection.getSubmitters())) {
-                    uncacheEntity(collection.getSubmitters());
-                }
                 if(Hibernate.isInitialized(collection.getTemplateItem())) {
                     uncacheEntity(collection.getTemplateItem());
                 }
-                if(Hibernate.isInitialized(collection.getWorkflowStep1())) {
-                    uncacheEntity(collection.getWorkflowStep1());
-                }
-                if(Hibernate.isInitialized(collection.getWorkflowStep2())) {
-                    uncacheEntity(collection.getWorkflowStep2());
-                }
-                if(Hibernate.isInitialized(collection.getWorkflowStep3())) {
-                    uncacheEntity(collection.getWorkflowStep3());
-                }
             }
 
+            if(!readOnlyEnabled && getSession().isDirty()) {
+                getSession().flush();
+            }
```

Comment:
On Slack #dev channel today, @ssolim noted these three lines now result in a `StaleStateException` once this PR is applied. See: https://dspace-org.slack.com/archives/C3SG47SGY/p1501833445027369

The solution seemed to be commenting out these lines (which were introduced in the related #1812 PR). Maybe this session flush isn't needed once the extra `uncacheEntity()` calls are removed??